plugins/inputs/monit/monit.go

```diff
@@ -0,0 +1,204 @@
+package monit
+
+import (
+	"encoding/xml"
+	"fmt"
+	"net/http"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"golang.org/x/net/html/charset"
+)
+
+var pendingActions = []string{"ignore", "alert", "restart", "stop", "exec", "unmonitor", "start", "monitor"}
+
+type Status struct {
+	Server   Server    `xml:"server"`
+	Platform Platform  `xml:"platform"`
+	Services []Service `xml:"service"`
+}
+
+type Server struct {
+	ID      string `xml:"id"`
+	Version string `xml:"version"`
+	Uptime  int    `xml:"uptime"`
+	Poll    int    `xml:"poll"`
+}
+
+type Platform struct {
+	Name    string `xml:"name"`
+	Release string `xml:"release"`
+	Version string `xml:"version"`
+	Machine string `xml:"machine"`
+	CPU     int    `xml:"cpu"`
+	Memory  int    `xml:"memory"`
+	Swap    int    `xml:"swap"`
+}
+
+type Service struct {
+	Type             string `xml:"type,attr"`
+	Name             string `xml:"name"`
+	Status           int64  `xml:"status"`
+	MonitoringStatus int64  `xml:"monitor"`
+	PendingAction    int    `xml:"pendingaction"`
+	Uptime           int64  `xml:"uptime"`
+	Memory           Memory `xml:"memory"`
+	CPU              CPU    `xml:"cpu"`
+	System           System `xml:"system"`
+}
+
+type Memory struct {
+	Percent       float64 `xml:"percent"`
+	PercentTotal  float64 `xml:"percenttotal"`
+	Kilobyte      int64   `xml:"kilobyte"`
+	KilobyteTotal int64   `xml:"kilobytetotal"`
+}
+
+type CPU struct {
+	Percent      float64 `xml:"percent"`
+	PercentTotal float64 `xml:"percenttotal"`
+}
+
+type System struct {
+	Load struct {
+		Avg01 float64 `xml:"avg01"`
+		Avg05 float64 `xml:"avg05"`
+		Avg15 float64 `xml:"avg15"`
+	} `xml:"load"`
+	CPU struct {
+		User   float64 `xml:"user"`
+		System float64 `xml:"system"`
+		Wait   float64 `xml:"wait"`
+	} `xml:"cpu"`
+	Memory struct {
+		Percent  float64 `xml:"percent"`
+		Kilobyte int64   `xml:"kilobyte"`
+	} `xml:"memory"`
+	Swap struct {
+		Percent  float64 `xml:"percent"`
+		Kilobyte float64 `xml:"kilobyte"`
+	} `xml:"swap"`
+}
+
+type Monit struct {
+	Address           string
+	BasicAuthUsername string
+	BasicAuthPassword string
+}
+
+func (m *Monit) Description() string {
+	return "Read metrics and status information about processes managed by Monit"
+}
+
+var sampleConfig = `
+  ## Monit
+  address = "http://127.0.0.1:2812"
+  basic_auth_username = ""
+  basic_auth_password = ""
+`
+
+func (m *Monit) SampleConfig() string {
+	return sampleConfig
+}
+
+func (m *Monit) Gather(acc telegraf.Accumulator) error {
+	client := &http.Client{}
```

Comment:
Perhaps make `client` part of the parent `Monit` object to prevent instantiating it each `Gather()`